terraform/aptos-node/gcp/README.md

```diff
@@ -0,0 +1,180 @@
+  ```
+  $ gsutil mb gs://BUCKET_NAME
+  # for example
+  $ gsutil mb gs://<project-name>-aptos-terraform-dev
+  ```
+
+3. Create Terraform file called `main.tf` in your working directory:
+  ```
+  $ cd ~/$WORKSPACE
+  $ touch main.tf
+  ```
+
+4. Modify `main.tf` file to configure Terraform, and create fullnode from Terraform module. Example content for `main.tf`:
+  ```
+  terraform {
+    required_version = "~> 1.1.0"
+    backend "gcs" {
+      bucket = "BUCKET_NAME" # bucket name created in step 2
+      prefix = "state/aptos-node"
+    }
+  }
+
+  module "aptos-node" {
+    # download Terraform module from aptos-labs/aptos-core repo
+    source        = "github.com/aptos-labs/aptos-core.git//terraform/aptos-node/gcp?ref=testnet"
+    region        = "us-central1"  # Specify the region
+    zone          = "c"            # Specify the zone suffix
+    project       = "<GCP Project Name>" # Specify your GCP project name
+    era           = 1              # bump era number to wipe the chain
+    image_tag     = "testnet" # Specify the docker image tag to use
+    validator_name = "<Name of Your Validator, no space>"
+  }
+  ```
+
+5. Initialise Terraform in the same directory of your `main.tf` file
+  ```
+  $ terraform init
+  ```
+This should download all the terraform dependencies for you, in the `.terraform` folder.
+
+6. Create a new Terraform workspace to isolate your environments:
+  ```
+  $ terraform workspace new $WORKSPACE
+  # This command will list all workspaces
+  $ terraform workspace list
+  ```
+
+7. Apply the configuration.
+
+  ```
+  $ terraform apply
+  ```
+
+  This might take a while to finish (10 - 20 minutes), Terraform will create all the resources on your cloud account. 
+
+8. Once terraform apply finishes, you can check if those resources are created:
+
+    - `kubectl get pods` this should have haproxy, validator and fullnode. with validator and fullnode pod `pending` (require further action in later steps)
+    - `kubectl get svc` this should have `validator-lb` and `fullnode-lb`, with an external-IP you can share later for connectivity.
+
+9. Get your node IP info:
+
+    ```
+    $ export VALIDATOR_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-validator-lb --output jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+
+    $ export FULLNODE_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-fullnode-lb --output jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+    ```
+
+10. Generate key pairs (node owner key, consensus key and networking key) in your working directory.
+
+    ```
+    $ aptos genesis generate-keys --output-dir ~/$WORKSPACE
+    ```
+
+    This will create a `private-keys.yaml` file for you, Backup your key files somewhere safe, this is important for you to establish ownership of your node, and it will be used to claim your rewards later if eligible. Very important!!
```

Comment:
1. comma → period.
2. `**IMPORTANT**: Backup your key files somewhere safe. These key files are important for you to establish ownership of your node, and you will use this information to claim your rewards later if eligible.`